plasmapy/simulation/particle_tracker/save_routines.py

```diff
@@ -0,0 +1,215 @@
+"""
+Module containing save routines for the particle tracker.
+"""
+
+__all__ = [
+    "AbstractSaveRoutine",
+    "DoNotSaveSaveRoutine",
+    "SaveOnceOnCompletion",
+    "IntervalSaveRoutine",
+]
+
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+import astropy.units as u
+import h5py
+
+
+class AbstractSaveRoutine(ABC):
+    """Abstract base class containing the necessary methods for a
+    `~plasmapy.simulation.particle_tracker.particle_tracker.ParticleTracker` save routine.
+
+    The save routine class is responsible for defining the conditions and hooks
+    for saving.
+
+    Parameters
+    ----------
+    output_directory : `~pathlib.Path`, optional
+        Output for objects that are saved to disk. If a directory is not specified
+        then a memory save routine is used.
+
+
+    Notes
+    -----
+    After every push, the `post_push_hook` method is called with the
+    respective `~plasmapy.simulation.particle_tracker.particle_tracker.ParticleTracker` object passed as a parameter.
+    Then, the hook calls `save_now` to determine whether or not the simulation state should be saved.
+    """
+
+    def __init__(self, output_directory: Path | None = None) -> None:
+        self.output_directory = output_directory
+
+        self._results = {}
+        self._quantities = {
+            "x": (u.m, "dataset"),
+            "v": (u.m / u.s, "dataset"),
+        }
+
+        self._particle_tracker = None
+
+    @property
+    def tracker(self):
+        """Return the `ParticleTracker` object for this stop condition."""
```

Comment:
I think the problem is here is `ParticleTracker` set up in the reference file? I believe the "WARNING: py:obj" error in building the docs is trying to refer to the `ParticleTracker` and it doesn't exist?

Reply:
Ah, I see! So even without the tilde
```
`ParticleTracker`
```
Is a reference to an object by that name, which is no longer in the same file! The error should go away if we did a string literal 

```
``ParticleTracker``
```

But it would be better to add `|ParticleTracker|` back to the `_global_substitutions` file, and then just use that throughout all of the docstrings. 

Does that make sense @JaydenR2305 ?